.github/workflows/release.yml

```diff
@@ -2,34 +2,31 @@ name: Release
 
 on:
   push:
-    tags:
-      - '*'
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
```

Comment:
~I get why you'd want to cancel in-progress merges, but is there a chance that this would break the order of history entries?~ never mind, thinking about this it seems like it should be fine. In the worst case, the release branch might not have the most recent code from main, but it'll be fixed with the next push.

Reply:
> concurrency

My goal is more to avoid any potential race condition if one build took longer than the later one, and consider it more important that `release` have the most recent build over all other concerns. I think if two commits are merged in quick succession, there's usually little benefit to generating intermediate builds. Open to disagreement though!

Reply:
> `fetch_depth: 0`

Because I also specify `ref: ${{ inputs.release_branch }}`, the checkout action will not pull all other branches. (I didn't feel we'd need them).